packages/squiggle-lang/src/rescript/utility/Sparklines.res

```diff
@@ -0,0 +1,51 @@
+// Port of Sindre Sorhus' Sparkly to Rescript
+// reference implementation: https://github.com/sindresorhus/sparkly
+// Omitting rgb "fire" style, so no `chalk` dependency
+
+type sparklyConfig = {
+  minimum: option<float>,
+  maximum: option<float>
+}
+
+let sparkly = (
+  numbers: list<float>,
+  ~options = {minimum: None, maximum: None}
+) => {
+  // if not numbers is not an array, throw typeerror "Expected an array"
+
+  // Unlike reference impl, we assume that all numbers are finite, i.e. no NaN.
+  let ticks = ["▁", "▂", "▃", "▄", "▅", "▆", "▇", "█"]
+  let minimum = switch options.minimum {
+    | None => Js.Math.minimum(numbers)
+    | Some(x) => x
+  }
+  let maximum = switch options.maximum {
+    | None => Js.Math.maximum(numbers)
+    | Some(x) => x
```

Comment:
I think this is just the ``default`` pattern. ``E.O.default(Js.Math.maximum(numbers))``

Reply:
noted, I'll try that out

Reply:
implemented in https://github.com/QURIresearch/squiggle/pull/124/commits/517a9128e2836fce30c73692a9dc44e0fcea3223